airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/DestinationHandler.java

```diff
@@ -4,37 +4,15 @@
 
 package io.airbyte.integrations.base.destination.typing_deduping;
 
-import java.time.Instant;
-import java.util.LinkedHashMap;
 import java.util.List;
-import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 
-public interface DestinationHandler<DialectTableDefinition> {
+public interface DestinationHandler<TableDefinition> {
 
-  Optional<DialectTableDefinition> findExistingTable(StreamId id) throws Exception;
-
-  /**
-   * Given a list of stream ids, return a map of stream ids to existing tables. If the table is
-   * missing, the key should not be present in the map.
-   *
-   * @param streamIds
-   * @return
-   * @throws Exception
-   */
-  LinkedHashMap<String, DialectTableDefinition> findExistingFinalTables(List<StreamId> streamIds) throws Exception;
-
-  boolean isFinalTableEmpty(StreamId id) throws Exception;
-
-  /**
-   * Returns the highest timestamp such that all records with _airbyte_extracted equal to or earlier
-   * than that timestamp have non-null _airbyte_loaded_at.
-   * <p>
-   * If the raw table is empty or does not exist, return an empty optional.
-   */
-  InitialRawTableState getInitialRawTableState(StreamId id) throws Exception;
+  void execute(final Sql sql) throws Exception;
 
-  record InitialRawTableState(boolean hasUnprocessedRecords, Optional<Instant> maxProcessedTimestamp) {}
+  List<CompletableFuture<DestinationInitialState<TableDefinition>>> gatherInitialState(List<StreamConfig> streamConfigs);
 
-  void execute(final Sql sql) throws Exception;
+  boolean existingSchemaMatchesStreamConfig(StreamConfig streamConfig, TableDefinition tableDefinition);
```

Comment:
do we still need this method at the top level interface? since DestinationInitialState has the isSchemaMismatch method inherently

(i.e. just move this method decl down to JdbcDestinationHandler)